mytoyota/models/endpoints/trip.py

```diff
@@ -0,0 +1,95 @@
+from pydantic import BaseModel, Field
+from datetime import date, datetime
+from uuid import UUID
+
+
+class _Pagination(BaseModel):
```

Comment:
Personally, I would do without the "_" prefixes everywhere. These are not really private classes, as the elements of the subclasses can be accessed at any time via the interface of the "Trips" class.

Reply:
Ok, forget what I wrote. That was complete nonsense. Of course we don't use the classes anywhere outside of trips. But I would have been more explicit in the naming to make it clear that these are models:

```python
from pydantic import BaseModel, Field
from datetime import date, datetime
from uuid import UUID
from typing import Optional


class _PaginationModel(BaseModel):
    currentPage: int
    limit: int
    offset: int
    pageCount: int
    previousOffset: int
    totalCount: int


class _SortedByModel(BaseModel):
    field: str
    order: str


class _MetaDataModel(BaseModel):
    pagination: _PaginationModel
    sortedBy: list[_SortedByModel]


class _ScoresModel(BaseModel):
    acceleration: int = Field(ge=0, le=100)
    advice: int = Field(ge=0, le=100, default=0)
    braking: int = Field(ge=0, le=100)
    constantSpeed: int = Field(ge=0, le=100, default=0)
    global_: Optional[int] = Field(ge=0, le=100, alias="global", default=None)


class _SummaryModel(BaseModel):
    countries: list[str]
    duration: int
    durationHighway: int
    durationIdle: int
    durationOverspeed: int
    fuelConsumption: float
    length: int
    lengthHighway: int
    lengthOverspeed: int
    maxSpeed: float


class _HDCModel(BaseModel):
    # Depending on
    #    car not being EV
    #    car being ev only
    #    car being hybrid but only used ev/fuel
    chargeDist: int = 0
    chargeTime: int = 0
    ecoDist: int = 0
    ecoTime: int = 0
    evDistance: int = 0
    evTime: int = 0
    powerDist: int = 0
    powerTime: int = 0


class _MonthSummary(BaseModel):
    hdc: Optional[_HDC] = None  # Only available on EV cars
    # histograms not imported
    month: int = Field(..., ge=1, le=12)
    scores: _ScoresModel
    summary: _SummaryModel
    year: int


class _TripSummaryModel(_SummaryModel):
    averageSpeed: float
    endLat: float
    endLon: float
    endTs: datetime
    nightTrip: bool
    startLat: float
    startLon: float
    startTs: datetime


class _TripModel(BaseModel):
    # behaviours not imported
    category: int
    hdc: Optional[_HDCModel] = None  # Only available on EV cars
    id: UUID
    scores: _ScoresModel
    summary: _TripSummaryModel


class TripsModel(BaseModel):
    metadata_: _MetaDataModel
    from_: date = Field(..., alias="from")
    to: date
    summary: list[_MonthSummaryModel] = []
    trips: list[_TripModel] = []

```

One more question, wouldn't it be better to use snake case names for the field names instead of the original camel case names? In my opinion, that would be more in line with the standards here. As you said, we don't want to reproduce the original interface 1:1, but rather build our own as an intermediate layer.

Reply:
This might be my lack of knowledge with pydantic but I was under the impression that the field name was the name of the JSON field parsed into it. Would this mean using an alias for all the ones that are in camel case?

Reply:
Exactly! These are the field names with which the JSON fields are parsed. This means that we either prepare the data accordingly or, as you rightly say, assign aliases to it.